hnn_core/cell.py

```diff
@@ -323,7 +344,6 @@ def insert_dipole(self, yscale):
             pos_all = np.array([seg.x for seg in sect.allseg()])
             # diff in yvals, scaled against the pos np.array. y_long as
             # in longitudinal
-            sect_name = sect.name().split('_', 1)[1]
             y_scale = (yscale[sect_name] * sect.L) * pos_all
```

Comment:
This (`y_scale` and `yscale`) is just unfair! We have to change these variable names while we're at it :)

```python
# Dipoles are oriented along the section. We record the projection of the
# current dipole moment onto the direction of the apical dendrite
dip_proj = (apical_projs[sect_name] * sect.L) * pos_all
```